### PR TITLE
docs: add documentation for tools

### PR DESCRIPTION
**Pull request by kurtisvg**
Adds more detailed documentation on tools and parameters. 

Only covers `postgres-generic` right now, looking into seeing if I can eliminate the other two tools.

### CODE REVIEW

File: docs/tools/README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Tools
+
+Tools represent an action your agent can take, such as running a SQL statement.
+You can define Tools as a map in the `sources` section of your `tools.yaml`
+file. Typically, a tool will require a source to act on:
+
+```yaml
+tools:
+ search_flights_by_number:
+    kind: cloud-sql-postgres-generic
```
Comment (Yuan325) [inline docs/tools/README.md:10]: We're going to remove this eventually (to make it the generic `postgres-sql` tool)? Or we will do those changes in another PR?

File: docs/tools/README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      - name: flight_number
+        type: string
+        description: 1 to 4 digit number
+```
+
+## Kinds of Tools
+
+We currently support the following types of kinds of tools:
+
+* [alloydb-postgres](./alloydb-pg.md) - Run a SQL statement against an AlloyDB
```
Comment (Yuan325) [inline docs/tools/README.md:49]: Same with this, we are removing this?

File: docs/tools/README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            type: string
+            description: Name of the airline. 
+```
+
+| **field**   | **type** | **required** | **description**                                                            |
+|-------------|:--------:|:------------:|----------------------------------------------------------------------------|
+| name        |  string  |     true     | Name of the parameter.                                                     |
+| type        |  string  |     true     | Must be "array"                                                            |
+| description |  string  |     true     | Natural language description of the parameter to describe it to the agent. |
+| items       |  string  |     true     | Specify a Parameter object for the type of the values in the array.        |
```
Comment (Yuan325) [inline docs/tools/README.md]: Seems like this is `array` type instead of string?

File: docs/tools/postgres-generic.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+PostgreSQL Generic Tools require one of the following sources:
+- [postgres](../sources/postgres.md)
+
+## Example
+
+```yaml
+tools:
+ search_flights_by_number:
+    kind: cloud-sql-postgres-generic
```
Comment (Yuan325) [inline docs/tools/postgres-generic.md]:
```suggestion
    kind: postgres-generic
```

File: docs/tools/postgres-generic.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+## Reference
+
+| **field**   | **type** | **required** | **description**                                                                          |
+|-------------|:--------:|:------------:|------------------------------------------------------------------------------------------|
+| kind        |  string  |     true     | Must be "postgres-generic".                                                              |
+| source      |  string  |     true     | Name of the source the SQL should execute on.                                            |
+| description |  string  |     true     | Port to connect to (e.g. "5432")                                                         |
+| statement   |  string  |     true     | SQL statement to execute on.                                                             |
+| parameters  |  string  |     true     | List of [parameters](README.md#specifying-parameters) that will be inserted into the SQL statement. |
```
Comment (Yuan325) [inline docs/tools/postgres-generic.md]: is the parameters type array or string?

File: docs/tools/README.md (hunk 1 above)
Comment (kurtisvg) [reply docs/tools/README.md:10]: Right -- I made those changes to `postgres-sql` in #43
